tools/hdf_dump_translation_dataset.py

```diff
@@ -0,0 +1,280 @@
+    self._tag_dtype = "S{}".format(max_tag_length)
+    self.hdf_file.create_dataset("seqTags", (self.number_of_lines,), dtype=self._tag_dtype)
+
+    self.hdf_file.create_dataset("inputs", (self.data_buffer_size,), maxshape=(None,),
+                                 dtype="int32", compression=self.compression_method)
+    self.hdf_file["targets/data"].create_dataset("classes", (self.data_buffer_size,), maxshape=(None,),
+                                                 dtype="int32", compression=self.compression_method)
+
+  def _write_attributes(self):
+    self.hdf_file["targets/size"].attrs["classes"] = (self._vocabulary_sizes["target"], 1)
+
+    # Those should be deprecated, but include nevertheless to exactly reproduce hdf_dump.
+    self.hdf_file.attrs["inputPattSize"] = self._vocabulary_sizes["source"]
+    self.hdf_file.attrs["numLabels"] = self._vocabulary_sizes["target"]
+
+  def _write_labels(self):
+    sorted_vocabulary_tuples = sorted(self._vocabularies["target"].items(), key=itemgetter(1))
+    labels = [word.encode("utf8") for (word, _) in sorted_vocabulary_tuples]
+
+    assert len(labels) == self._vocabulary_sizes["target"], "Word ids were not unique."
+
+    max_label_length = max([len(label) for label in labels])
+    dtype = "S{}".format(max_label_length + 1)
+
+    labels = [numpy.array(label, dtype=dtype, ndmin=1) for label in labels]
+    labels = numpy.concatenate(labels)
+
+    self.hdf_file["targets/labels"].create_dataset("classes", (self._vocabulary_sizes["target"],),
+                                                   data=labels, dtype=dtype)
+
+  def _write_data(self):
+    source_chunk, target_chunk = self._get_chunk()
+
+    if not source_chunk:
+      self._finalize_data()
+      return True
+
+    source_lengths = [len(line) for line in source_chunk]
+    target_lengths = [len(line) for line in target_chunk]
+
+    chunk_size = len(source_chunk)
+
+    self._write_lengths(source_lengths, target_lengths)
+    self._write_tags(chunk_size)
+
+    self._write_data_indices(source_chunk, "source")
+    self._write_data_indices(target_chunk, "target")
+
+    self._number_of_processed_lines += chunk_size
+
+    return False
+
+  def _get_chunk(self):
+    source_lines = islice(self.source_file_handle, self.line_buffer_size)
+    target_lines = islice(self.target_file_handle, self.line_buffer_size)
+
+    source_chunk = []
+    target_chunk = []
+    for source_line, target_line in zip_longest(source_lines, target_lines):
+      assert source_line is not None and target_line is not None, "Number of source and target lines differ."
+
+      source_indices = self._line_to_indices(source_line, "source")
+      target_indices = self._line_to_indices(target_line, "target")
+
+      source_chunk.append(source_indices)
+      target_chunk.append(target_indices)
+
+    return source_chunk, target_chunk
+
+  def _write_lengths(self, source_lengths, target_lengths):
```

Comment:
Arguments not documented here.

Reply:
Also, if these are numpy arrays (I don't know...), you should document the dtype and shape.